CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -262,5 +262,14 @@ private List<long> GetStudentsIDNotIncludeInGroup(IList<VisitDto> visit, IList<l
             return result;
         }
 
+        public async Task<IList<LessonDto>> GetLessonsForStudentAsync(FilterLessonsRequestDto filterModel)
+        {
+            long accountId = _currentUserService.AccountId;
+            var student = await _unitOfWork.StudentRepository.GetStudentByAccountIdAsync(accountId);
+
+            var lessonsForStudent = await _unitOfWork.LessonRepository.GetLessonsForStudentAsync(filterModel.StudentGroupId, filterModel.StartDate, filterModel.FinishDate, student.Id);
```

Comment:
```suggestion
            var lessonsForStudent = await _unitOfWork.LessonRepository.GetLessonsForStudentAsync(filterModel.StudentGroupId, filterModel.StartDate, filterModel.FinishDate, _currentUserService.EntityId);
```
By the moment the code execution enters this method, we already have student id of a user. It's an `EntityId`  property inside `_currentUserService`.

And by the current implementation of this method **we already assuming that user that made a request is in Student role**. Same inside controller:
![asdasd](https://user-images.githubusercontent.com/38729881/109419949-a7338380-79d8-11eb-867f-498a7e65d634.png)

So why make additional database call to get student id that we already have? (lines 267 & 268)

@mahotskuy 
I wonder if there's any point in validation of student id obtained from `HttpContext` (with `_currentUserService.EntityId`) inside this service method (given that this service method will be covered with unit tests).

Reply:
Solved in https://github.com/ita-social-projects/WhatBackend/pull/438